tss-esapi/src/error/return_code/tpm/format_zero/warning.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use crate::{constants::return_code::TpmFormatZeroWarning, Error, Result};
+use std::convert::TryFrom;
+
+/// Type representing the TPM format zero warning
+/// response code.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct TpmFormatZeroWarningResponseCode {
+    error_number: TpmFormatZeroWarning,
+}
+
+impl TpmFormatZeroWarningResponseCode {
+    /// Returns the error number associated with the TPM
+    /// format zero warning.
+    pub const fn error_number(&self) -> TpmFormatZeroWarning {
```

Comment:
Yes, I am not to fond of it but I did it this way because in the specification of the structures the area which contain the error code is referred to "error number" no matter if it is a warning or not.

Reply:
But I guess we could change the name to something more generic as "kind" or just "number".